base/dates/periods.jl

```diff
@@ -443,24 +443,35 @@ Base.promote_rule(::Type{Year}, ::Type{Month}) = Month
 (==)(x::FixedPeriod, y::OtherPeriod) = throw(MethodError(==, (x, y)))
 (==)(x::OtherPeriod, y::FixedPeriod) = throw(MethodError(==, (x, y)))
 
+const fixedperiod_seed = UInt === UInt64 ? 0x5b7fc751bba97516 : 0xeae0fdcb
+const otherperiod_seed = UInt === UInt64 ? 0xe1837356ff2d2ac9 : 0x170d1b00
+# Use Int128 so that even Week(typemax(Int64)) or Week(typemin(Int64))
+# can be converted to nanoseconds and Year(typemax(Int64)) or Year(typemin(Int64))
+# can be converted to months without overflow
```

Comment:
I think for this PR you can just change `tons` to always return a `Int128`. You can leave `toms` and `days` as they were. I don't especially like that `toms` may return either a `Int64` or a `Float64` but we can address that later.